client/src/app/site/projector/components/presentation-control/presentation-control.component.ts

```diff
@@ -25,15 +26,34 @@ export class PresentationControlComponent extends BaseComponent {
     @Input()
     public set projector(projector: ViewProjector) {
         this._projector = projector;
-        this.updateElements();
+        if (projector?.id !== this.oldProjectorId) {
+            this.oldProjectorId = projector?.id;
+
+            if (this.projectionSubscription) {
+                this.projectionSubscription.unsubscribe();
+                this.projectionSubscription = null;
+            }
+            if (this.projector) {
+                this.projectionSubscription = this.projectionRepo
+                    .getGeneralViewModelObservable()
+                    .subscribe(projection => {
+                        if (projection?.current_projector_id === this.projector.id) {
+                            this.updateProjections();
+                        }
+                    });
```

Comment:
I suppose this part can, as well, heavily simplified after projections can be accessed as observables, thus we might wait with merging that

Reply:
Right, same "problem" here